src/components/Logout/LogoutScreen.test.tsx

```diff
@@ -0,0 +1,298 @@
+import { render, waitFor } from "@testing-library/react-native";
+import React, { FunctionComponent } from "react";
+import { NavigationScreenProp, withNavigation } from "react-navigation";
+import { AlertModalContext } from "../../context/alert";
+import { AuthStoreContext } from "../../context/authStore";
+import { CampaignConfigsStoreContext } from "../../context/campaignConfigsStore";
+import { AppMode, ConfigContext } from "../../context/config";
+import { ImportantMessageSetterContext } from "../../context/importantMessage";
+import { callLogout, LogoutError } from "../../services/auth";
+import { NetworkError } from "../../services/helpers";
+import { AuthCredentials } from "../../types";
+import { Sentry } from "../../utils/errorTracking";
+import { ErrorBoundary } from "../ErrorBoundary/ErrorBoundary";
+import { LogoutScreen } from "./LogoutScreen";
+
+jest.mock("../../utils/errorTracking");
+jest.mock("../../services/auth", () => ({
+  ...jest.requireActual("../../services/auth"),
+  callLogout: jest.fn(),
+}));
+jest.mock("react-navigation", () => ({
+  // eslint-disable-next-line react/display-name
+  withNavigation: (Component: FunctionComponent) => (props: any) => (
+    <Component navigation={{ navigate: mockNavigate }} {...props} />
+  ),
+  // eslint-disable-next-line react/display-name
+  withNavigationFocus: (Component: FunctionComponent) => (props: any) => (
+    <Component navigation={{ navigate: mockNavigate }} {...props} />
+  ),
+}));
+
+const mockCaptureException = Sentry.captureException as jest.MockedFunction<
+  typeof Sentry.captureException
+>;
+const mockNavigate = jest.fn() as jest.MockedFunction<
+  NavigationScreenProp<unknown>["navigate"]
+>;
+const mockCallLogout = callLogout as jest.MockedFunction<typeof callLogout>;
+
+const LogoutScreenContainer = withNavigation(LogoutScreen);
+
+const keyA = "keyA";
+const keyB = "keyB";
+const keyC = "keyC";
+
+const authA: AuthCredentials = {
+  endpoint: "endA",
+  expiry: 0,
+  operatorToken: "optA",
+  sessionToken: "sesA",
+};
+const authB: AuthCredentials = {
+  endpoint: "endB",
+  expiry: 0,
+  operatorToken: "optB",
+  sessionToken: "sesB",
+};
+const authC: AuthCredentials = {
+  endpoint: "endC",
+  expiry: 0,
+  operatorToken: "optC",
+  sessionToken: "sesC",
+};
+
+const authCredentials = {
+  [keyA]: authA,
+  [keyB]: authB,
+  [keyC]: authC,
+};
+
+const defaultAuthStoreContextValue = {
+  authCredentials: authCredentials,
+  clearAuthCredentials: jest.fn(),
+  hasLoadedFromStore: true,
+  removeAuthCredentials: jest.fn(),
+  setAuthCredentials: jest.fn(),
+};
+
+const defaultCampaignConfigStoreContextValue = {
+  allCampaignConfigs: {},
+  clearCampaignConfigs: jest.fn(),
+  hasLoadedFromStore: true,
+  removeCampaignConfig: jest.fn(),
+  setCampaignConfig: jest.fn(),
+};
+
+const defaultConfigContextProviderValue = {
```

Comment:
Try to create these variables this way instead

```
describe(‘test’, () => {
  let var1;
  let var2;
  let …

  beforeAll(() => {
    var1 = …;
    var2 = …;
    …
  });
});
```

The reason is that all the statements outside of the before block will be executed regardless of whether this test should be executed. I.e. if I were just running a single test file A, all the variables initialised outside of the `beforeAll` block in test file B will also be run. This unnecessarily increases the overhead of running tests.